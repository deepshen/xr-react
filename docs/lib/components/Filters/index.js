!function(e,t){if("object"==typeof exports&&"object"==typeof module)module.exports=t(require("react"),require("classnames"),require("react-dom"));else if("function"==typeof define&&define.amd)define(["react","classnames","react-dom"],t);else{var n="object"==typeof exports?t(require("react"),require("classnames"),require("react-dom")):t(e.React,e.classnames,e.ReactDOM);for(var r in n)("object"==typeof exports?exports:e)[r]=n[r]}}(window,function(n,r,o){return function(c){function e(e){for(var t,n,r=e[0],o=e[1],a=e[2],u=0,l=[];u<r.length;u++)n=r[u],Object.prototype.hasOwnProperty.call(f,n)&&f[n]&&l.push(f[n][0]),f[n]=0;for(t in o)Object.prototype.hasOwnProperty.call(o,t)&&(c[t]=o[t]);for(s&&s(e);l.length;)l.shift()();return p.push.apply(p,a||[]),i()}function i(){for(var e,t=0;t<p.length;t++){for(var n=p[t],r=!0,o=1;o<n.length;o++){var a=n[o];0!==f[a]&&(r=!1)}r&&(p.splice(t--,1),e=u(u.s=n[0]))}return e}var n={},f={2:0},p=[];function u(e){if(n[e])return n[e].exports;var t=n[e]={i:e,l:!1,exports:{}};return c[e].call(t.exports,t,t.exports,u),t.l=!0,t.exports}u.m=c,u.c=n,u.d=function(e,t,n){u.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:n})},u.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},u.t=function(t,e){if(1&e&&(t=u(t)),8&e)return t;if(4&e&&"object"==typeof t&&t&&t.__esModule)return t;var n=Object.create(null);if(u.r(n),Object.defineProperty(n,"default",{enumerable:!0,value:t}),2&e&&"string"!=typeof t)for(var r in t)u.d(n,r,function(e){return t[e]}.bind(null,r));return n},u.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return u.d(t,"a",t),t},u.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},u.p="";var t=window.webpackJsonp=window.webpackJsonp||[],r=t.push.bind(t);t.push=e,t=t.slice();for(var o=0;o<t.length;o++)e(t[o]);var s=r;return p.push([47,0]),i()}({0:function(e,t){e.exports=n},15:function(e,t){e.exports=o},4:function(e,t){e.exports=r},47:function(e,t,n){"use strict";n.r(t);var s=n(0),d=n.n(s),r=n(58),y=n.n(r),o=(n(83),n(42)),v=n.n(o),a=(n(84),n(45)),b=n.n(a),u=(n(70),n(33)),m=n.n(u),l=(n(61),n(40)),O=n.n(l),h=(n(85),function(){return(h=Object.assign||function(e){for(var t,n=1,r=arguments.length;n<r;n++)for(var o in t=arguments[n])Object.prototype.hasOwnProperty.call(t,o)&&(e[o]=t[o]);return e}).apply(this,arguments)}),g=function(e,t){var n={};for(var r in e)Object.prototype.hasOwnProperty.call(e,r)&&t.indexOf(r)<0&&(n[r]=e[r]);if(null!=e&&"function"==typeof Object.getOwnPropertySymbols)for(var o=0,r=Object.getOwnPropertySymbols(e);o<r.length;o++)t.indexOf(r[o])<0&&Object.prototype.propertyIsEnumerable.call(e,r[o])&&(n[r[o]]=e[r[o]]);return n},j=O.a.Option;t.default=function(e){function f(e,t){var n,r=(null===(n=null==e?void 0:e.target)||void 0===n?void 0:n.value)||e;p[t]=r,a(h({},p),t,r)}function t(){u()}var n=e.filters,r=void 0===n?[]:n,o=e.value,p=void 0===o?{}:o,a=e.onChange,u=e.onSearch,l=Object(s.useState)(!1),c=l[0],i=l[1];Object(s.useEffect)(function(){c&&"{}"===JSON.stringify(p)&&(t(),i(!1))},[c,p]);return d.a.createElement("div",null,d.a.createElement(y.a,{gutter:16},r.map(function(e,t){var n,r=e.name,o=e.type,a=e.options,u=void 0===a?[]:a,l=e.render,c=g(e,["name","type","options","render"]),i=null;switch(o){case"input":i=d.a.createElement(b.a,h({allowClear:!0,onChange:function(e){return f(e,r)},value:p[r]},c));break;case"select":i=d.a.createElement(O.a,h({allowClear:!0,value:p[r],onChange:function(e){return f(e,r)},style:{width:"100%"}},c),u.map(function(e){return d.a.createElement(j,{value:e.value,key:e.value},e.label)}));break;case"render":l&&(i="string"==typeof(n=l())?n:Object(s.cloneElement)(n,{value:p[r],onChange:function(e){return f(e,r)}}));break;default:i=null}return d.a.createElement(v.a,{key:e.name||t,span:4},i)}),d.a.createElement(v.a,null,d.a.createElement(m.a,{onClick:function(){a({}),i(!0)}},"重置"),d.a.createElement(m.a,{type:"primary",onClick:t},"搜索"))))}}})});