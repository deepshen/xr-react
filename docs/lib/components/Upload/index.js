!function(e,t){if("object"==typeof exports&&"object"==typeof module)module.exports=t(require("react"));else if("function"==typeof define&&define.amd)define(["react"],t);else{var n="object"==typeof exports?t(require("react")):t(e.React);for(var o in n)("object"==typeof exports?exports:e)[o]=n[o]}}(window,function(n){return function(l){function e(e){for(var t,n,o=e[0],r=e[1],i=e[2],u=0,a=[];u<o.length;u++)n=o[u],Object.prototype.hasOwnProperty.call(f,n)&&f[n]&&a.push(f[n][0]),f[n]=0;for(t in r)Object.prototype.hasOwnProperty.call(r,t)&&(l[t]=r[t]);for(d&&d(e);a.length;)a.shift()();return p.push.apply(p,i||[]),c()}function c(){for(var e,t=0;t<p.length;t++){for(var n=p[t],o=!0,r=1;r<n.length;r++){var i=n[r];0!==f[i]&&(o=!1)}o&&(p.splice(t--,1),e=u(u.s=n[0]))}return e}var n={},f={3:0},p=[];function u(e){if(n[e])return n[e].exports;var t=n[e]={i:e,l:!1,exports:{}};return l[e].call(t.exports,t,t.exports,u),t.l=!0,t.exports}u.m=l,u.c=n,u.d=function(e,t,n){u.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:n})},u.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},u.t=function(t,e){if(1&e&&(t=u(t)),8&e)return t;if(4&e&&"object"==typeof t&&t&&t.__esModule)return t;var n=Object.create(null);if(u.r(n),Object.defineProperty(n,"default",{enumerable:!0,value:t}),2&e&&"string"!=typeof t)for(var o in t)u.d(n,o,function(e){return t[e]}.bind(null,o));return n},u.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return u.d(t,"a",t),t},u.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},u.p="";var t=window.webpackJsonp=window.webpackJsonp||[],o=t.push.bind(t);t.push=e,t=t.slice();for(var r=0;r<t.length;r++)e(t[r]);var d=o;return p.push([274,6]),c()}({0:function(e,t){e.exports=n},274:function(e,t,n){"use strict";n.r(t);function k(){}var o=n(0),T=n.n(o),r=n(267),F=n.n(r);t.default=function(e){var t=e.multiple,n=void 0!==t&&t,o=e.authSuccess,r=void 0===o?k:o,i=e.authFailed,u=void 0===i?k:i,a=e.beforeUpload,l=void 0===a?k:a,c=e.uploadProgress,f=void 0===c?k:c,p=e.uploadFileSuccess,d=void 0===p?k:p,s=e.uploadFileFailed,v=void 0===s?k:s,h=e.uploadError,y=void 0===h?k:h,b=e.uploadFinish,g=void 0===b?k:b,m=e.onSuccess,x=void 0===m?k:m,w=e.isPrivate,j=void 0!==w&&w,S=e.onChange,O=void 0===S?function(){return!0}:S,P=new F.a({getAuth:function(){parseInt(String((new Date).valueOf()/1e3),10);return fetch("https://open.aihaisi.com/qcloud/api/token").then(function(e){return e.json()}).then(function(e){return{TmpSecretId:e.data.tmpSecretId,TmpSecretKey:e.data.tmpSecretKey,ExpiredTime:e.data.expiration,XCosSecurityToken:e.data.sessionToken}})},region:"ap-shanghai"});P.on("auth-success",function(e){r(e)}),P.on("auth-failed",function(e){u(e)}),P.on("before-upload",function(e){l(e)}),P.on("upload-progress",function(e){f(e)}),P.on("upload-file-success",function(e){d(e)}),P.on("upload-file-failed",function(e){v(e)}),P.on("upload-error",function(e){y(e)}),P.on("upload-finish",function(e){g(e)});return T.a.createElement("div",{style:{position:"relative",overflow:"hidden",display:"inline-block"}},e.children?e.children:T.a.createElement("div",{style:{display:"flex",alignItems:"center",justifyContent:"center",width:"80px",height:"80px",border:"1px solid #ececec",borderRadius:"6px"}},"+"),T.a.createElement("input",{multiple:n,onChange:function(a){var n=a.target.files;if(O())for(var l=[],e=0,t=n.length;e<t;e+=1)!function(r,i){var u,e=n[r],t=new FileReader;t.readAsArrayBuffer(e),t.onload=function(e){var t,n,o;u="object"==typeof e.target.result?new Blob([e.target.result]):a.target.result,l.push(u),r===i-1&&(t=j,n=l,o=x,P.upload(n,{bucket:t?"private-10000230":"public-10000230",urlDomain:t?"privimg.xingren.com":""}).then(function(e){o(e)}).finally(function(){}))}}(e,t)},style:{position:"absolute",left:"0",top:"0",width:"100%",height:"100%",opacity:0},type:"file"}))}}})});