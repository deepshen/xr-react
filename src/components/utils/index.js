export const handleAuth = () => fetch('https://open.aihaisi.com/qcloud/api/token').then((data) => data.json());

export const handleUpload = (upload, isPrivate, files, cb) => {
  upload.upload(files, {
    bucket: isPrivate ? 'private-10000230' : 'public-10000230',
    urlDomain: isPrivate ? 'privimg.xingren.com' : '',
  }).then((res) => {
    cb(res);
  }).finally(() => {

  });
};
