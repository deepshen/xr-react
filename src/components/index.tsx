import Upload from './lib/Upload'

export {Upload}
