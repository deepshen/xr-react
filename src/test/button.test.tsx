import React from 'react'
import {Upload} from '../components/index'
import * as renderer from 'react-test-renderer'


describe('upload compoennt',() => {
  it('正确渲染upload', () => {
    const json = renderer.create(<Upload />).toJSON();
    expect(json).toMatchSnapshot()
  })
})
