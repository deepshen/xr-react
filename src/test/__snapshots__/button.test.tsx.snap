// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`upload compoennt 正确渲染upload 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "border": "1px solid #ececec",
        "borderRadius": "6px",
        "display": "flex",
        "height": "80px",
        "justifyContent": "center",
        "width": "80px",
      }
    }
  >
    +
  </div>
  <input
    multiple={false}
    onChange={[Function]}
    style={
      Object {
        "height": "100%",
        "left": "0",
        "opacity": 0,
        "position": "absolute",
        "top": "0",
        "width": "100%",
      }
    }
    type="file"
  />
</div>
`;
